src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1457,8 +1455,10 @@ public void RemoveWorkspace(WorkspaceModel workspace)
         {
             if (_workspaces.Remove(workspace))
             {
-                if (workspace is HomeWorkspaceModel)
+                OnWorkspaceRemoveStarted(workspace);
```

Comment:
This is a bit like PreWorkspaceDisposed and I think it should be put just before the line where workspace is disposed.